rwengine/src/engine/Garage.cpp

```diff
@@ -0,0 +1,889 @@
+        }
+
+        case Type::BombShop1: {
+            break;
+        }
+
+        case Type::BombShop2: {
+            break;
+        }
+
+        case Type::BombShop3: {
+            break;
+        }
+
+        case Type::Respray: {
+            break;
+        }
+
+        case Type::CollectCars1:
+        case Type::CollectCars2: {
+            break;
+        }
+
+        case Type::MissionForCarToComeOut: {
+            break;
+        }
+
+        case Type::Crusher: {
+            break;
+        }
+
+        case Type::MissionKeepCar: {
+            break;
+        }
+
+        case Type::Hideout1:
+        case Type::Hideout2:
+        case Type::Hideout3: {
+            break;
+        }
+
+        case Type::MissionToOpenAndClose: {
+            break;
+        }
+
+        case Type::MissionForSpecificCar: {
+            break;
+        }
+
+        case Type::MissionKeepCarAndRemainClosed: {
+            break;
+        }
+
+        default: { break; }
+    }
+}
+
+void Garage::doOnCloseEvent() {
+    auto player = engine->getPlayer();
+    auto plyChar = player->getCharacter();
+    auto playerPosition = plyChar->getPosition();
+    auto playerVehicle = plyChar->getCurrentVehicle();
+    bool playerIsInVehicle = playerVehicle != nullptr;
+    RW_UNUSED(playerPosition);
+    RW_UNUSED(playerIsInVehicle);
+
+    switch (type) {
+        case Type::Mission: {
+            player->setInputEnabled(true);
+            break;
+        }
+
+        case Type::BombShop1:
+        case Type::BombShop2:
+        case Type::BombShop3: {
+            // Find out real value
+            garageTimer = engine->getGameTime() + 1.5f;
+
+            break;
+        }
+
+        case Type::Respray: {
+            // Find out real value
+            garageTimer = engine->getGameTime() + 2.f;
+            playerVehicle->setHealth(1000.f);
+
+            break;
+        }
+
+        case Type::CollectCars1:
+        case Type::CollectCars2: {
+            break;
+        }
+
+        case Type::MissionForCarToComeOut: {
+            break;
+        }
+
+        case Type::Crusher: {
+            break;
+        }
+
+        case Type::MissionKeepCar: {
+            break;
+        }
+
+        case Type::Hideout1:
+        case Type::Hideout2:
+        case Type::Hideout3: {
+            break;
+        }
+
+        case Type::MissionToOpenAndClose: {
+            break;
+        }
+
+        case Type::MissionForSpecificCar: {
+            break;
+        }
+
+        case Type::MissionKeepCarAndRemainClosed: {
+            break;
+        }
+
+        default: { break; }
+    }
+}
+
+void Garage::doOnStartOpeningEvent() {
+    auto player = engine->getPlayer();
+    auto plyChar = player->getCharacter();
+    auto playerPosition = plyChar->getPosition();
+    auto playerVehicle = plyChar->getCurrentVehicle();
+    bool playerIsInVehicle = playerVehicle != nullptr;
+    RW_UNUSED(playerPosition);
+    RW_UNUSED(playerIsInVehicle);
```

Comment:
See other comment about `playerIsInVehicle`. Not sure how useful creating all those variables is when there is no user (and it's unlikely there'll be one).
It's probably more important to make it obvious that we check for the valitity of playerVehicle when it's being used.

Reply:
Probably it would be used in the future for other garage types.